webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/Compliance.scala

```diff
@@ -301,6 +321,300 @@ object JsonCompliance {
     }
   }
 
+  implicit class JsonbyDirectiveCompliance(val directive: ByDirectiveCompliance) extends AnyVal {
+    def toJsonV6 = (
+      ("id"                    -> directive.id.serialize)
+        ~ ("name"              -> directive.name)
+        ~ ("compliance"        -> directive.compliance.complianceWithoutPending())
+        ~ ("complianceDetails" -> percents(directive.compliance, CompliancePrecision.Level2))
+        ~ ("nodes"             -> byNodes(directive.nodes.flatMap(_.nodeCompliances), 10, CompliancePrecision.Level2))
+      )
+
+    /*
+     * level:
+     * - up to 1 : rules,
+     * - 2: rules & directives
+     * - 3: rules, directives, components
+     * - 4 and up: rules, directives, components, node and component values
+     */
+
+    def toJson(level: Int, precision: CompliancePrecision) = (
+      ("id"                    -> directive.id.serialize)
+        ~ ("name"              -> directive.name)
+        ~ ("compliance"        -> directive.compliance.complianceWithoutPending(precision))
+        ~ ("mode"              -> directive.mode.name)
+        ~ ("complianceDetails" -> percents(directive.compliance, precision))
+//        ~ ("rules"             -> rules(directive.rules.map(_.), level, precision))
+        ~ ("nodes"             -> byNodes(directive.nodes, level, precision))
```

Comment:
I'm trying to get a structure like
```
"nodes": [
          {
            "id": "831b9c93-bb96-4d48-8c4e-91ca19b98bde",
            "name": "prod-zabbix-01.lab.rudder.io",
            "compliance": 100.0,
            "complianceDetails": {
              "auditCompliant": 100.0
            },
            "rules": [
              {
                "id": "94e9b2ed-6f5a-44a2-a8c5-82b88f4d85b7",
                "name": "Training - Trigger Command",
                "compliance": 100.0,
                "complianceDetails": {
                  "auditCompliant": 100.0
                },
                "components": [
                  {
                    "name": "Copy script agent-hello onto Nodes ",
                    "compliance": 100.0,
                    "complianceDetails": {
                      "auditCompliant": 100.0
                    },
                    "values": [
                      {
                        "value": "/opt/rudder/share/commands/agent-hello",
                        "reports": [
                          {
                            "status": "auditCompliant",
                            "message": "Copying /opt/rudder/share/commands/agent-hello from agent-hello was correct"
                          }
                        ]
                      }
                    ]
                  }
                ]
```

Reply:
why do you have a rule and not a directive?

Reply:
directive is at top level. we want compliance for a directive